File: Day8/d8.py
def isConsecutive(nums): 
    numSet = set(nums)
    longest = 0
    for num in numSet:
        if num-1 not in numSet:
            currentNum = num
            currentStreak = 1
            while currentNum+1 in numSet:
                currentNum += 1
                currentStreak += 1
            longest = max(longest, currentStreak)
    return longest
    
a = [4, 2, 1, 6, 5]
b = [100,4,200,1,3,2]
print(isConsecutive(b))
print(isConsecutive(a))

File: Day8/d8.js
const isConsecutive = (nums) => {
    // create a set from the array
    const set = new Set(nums);
    var longestStreak = 0;
    for (let i in nums) {
        if ( !set.has(i - 1)) {
            var currentNum = i;
            var currentStreak = 1;
            while (set.has(currentNum + 1)) {
                currentNum++;
                currentStreak++;
            }
            longestStreak = Math.max(longestStreak, currentStreak);
        }
        return longestStreak;
    }
}

var nums = [100, 4, 200, 1, 3, 2];
console.log(isConsecutive(nums));